src/Integration.Vsix.UnitTests/InfoBar/InfoBarManagerTests.cs

```diff
@@ -303,6 +303,62 @@ public void AttachInfoBarWithButtons_ButtonClickedEvent_RaisedWithTheRightButton
             eventHandler.VerifyNoOtherCalls();
         }
 
+        #region MainWindow tests
+        // The ToolWindow tests above cover the common implementation of gold bar functionality.
+        // These MainWindow tests just cover a few extra main window specific cases.
+
+        [TestMethod]
+        public void InfoBarManager_AttachInfoBarToMainWindowAndDetach_ArgChecks()
+        {
+            var testSubject = CreateTestSubject();
+
+            Action act = () => testSubject.AttachInfoBarToMainWindow(null, SonarLintImageMoniker.OfficialSonarLintMoniker);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("message");
+
+            act = () => testSubject.AttachInfoBarToMainWindow("a message", SonarLintImageMoniker.OfficialSonarLintMoniker, null);
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("buttonTexts");
+        }
+
+        [TestMethod]
+        public void InfoBarManager_AttachInfoBarToMainWindowAndDetach_AttachAndDetach()
```

Comment:
This is the test from your first commit. I appended `_AttachAndDetach` to the name, and added the assertion on line 335 that the returned infobar isn't null.

Reply:
Wouldn't it make more sense to have "_AttachesAndDetaches" instead? just because else reading Attach...Detach_AtachAndDetach is a bit confusing to me